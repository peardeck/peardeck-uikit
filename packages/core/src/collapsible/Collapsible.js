var __rest =
  (this && this.__rest) ||
  function(s, e) {
    var t = {};
    for (var p in s)
      if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
      for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
        if (
          e.indexOf(p[i]) < 0 &&
          Object.prototype.propertyIsEnumerable.call(s, p[i])
        )
          t[p[i]] = s[p[i]];
      }
    return t;
  };
import React, { useState } from "react";
import styled from "styled-components";
import { CollapsibleHeader } from "./CollapsibleHeader";
const StyledCollapsible = styled.div`
  border: 1px solid #eee;
`;
const StyledCollapsibleContent = styled.div`
  transition: all 0.5s;
  overflow: hidden;
  max-height: ${({ collapsed }) => (collapsed ? "0" : "600px")};
`;
export const Collapsible = _a => {
  var { children, collapsed, onCollapsedChange } = _a,
    props = __rest(_a, ["children", "collapsed", "onCollapsedChange"]);
  return React.createElement(
    StyledCollapsible,
    Object.assign({}, props),
    React.createElement(CollapsibleHeader, {
      title: "The Title",
      subTitle: "A subtitle goes here",
      collapsed: collapsed,
      onCollapsedChange: onCollapsedChange,
    }),
    React.createElement(
      StyledCollapsibleContent,
      { collapsed: collapsed },
      children
    )
  );
};
export const UncontrolledCollapsible = props => {
  const [collapsed, onCollapsedChange] = useState(false);
  return React.createElement(
    Collapsible,
    Object.assign({}, props, {
      collapsed: collapsed,
      onCollapsedChange: onCollapsedChange,
    })
  );
};
//# sourceMappingURL=Collapsible.js.map
