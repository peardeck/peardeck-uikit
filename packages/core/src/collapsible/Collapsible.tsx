import React, { useState } from "react";
import styled from "styled-components";
import { CollapsibleHeader, CollapsibleHeaderProps } from './CollapsibleHeader';

interface CollapsibleProps extends CollapsibleHeaderProps {
  /**
   * Called when the collapsed state should be changed.
   *
   * Signature:
   * ```
   * onCollapsedChanged(shouldBeCollapsed: boolean)
   * ```
   */
  onCollapsedChange: Function

  /**
   * The content to render in the collapsible content section.
   */
  children: React.ReactNode
}

const StyledCollapsible = styled.div`
  border: 1px solid #eee;
`

interface StyledCollapsibleContentProps {
  collapsed: boolean
}

const StyledCollapsibleContent = styled.div<StyledCollapsibleContentProps>`
  transition: all .5s;
  overflow: hidden;
  max-height: ${({ collapsed }: { collapsed: boolean }) => collapsed ? "0" : "600px"};
`;

export const Collapsible = ({ children, collapsed, onCollapsedChange, ...props }: CollapsibleProps) => (
  <StyledCollapsible {...props}>
    <CollapsibleHeader
      title='The Title'
      subTitle='A subtitle goes here'
      collapsed={collapsed}
      onCollapsedChange={onCollapsedChange}
    />
    <StyledCollapsibleContent collapsed={collapsed}>
      {children}
    </StyledCollapsibleContent>
  </StyledCollapsible>
);

interface UncontrolledCollapsibleProps {
  /**
   * Test!
   */
  title: string,
  /* Subtitle info goes here? */
  subTitle?: string
  children: React.ReactNode
}

export const UncontrolledCollapsible = (props: UncontrolledCollapsibleProps) => {
  const [collapsed, onCollapsedChange] = useState(false);
  return (<Collapsible {...props} collapsed={collapsed} onCollapsedChange={onCollapsedChange} />)
};
