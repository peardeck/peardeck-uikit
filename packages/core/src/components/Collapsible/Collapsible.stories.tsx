import React from "react";
import { useState } from "@storybook/addons";
import { Collapsible } from "./Collapsible";

export default {
  component: Collapsible,
  title: "core|Collapsible",
};

export const Default = () => {
  const [collapsed, onCollapsedChange] = useState(false);

  return (
    <Collapsible
      title="This is the title"
      subTitle="...and a subtitle goes here"
      collapsed={collapsed}
      onCollapsedChange={onCollapsedChange}
    >
      Content goes here!
      <div
        style={{
          height: "10em",
          border: "1px solid red",
          backgroundColor: "pink",
          color: "white",
        }}
      >
        Make room!
      </div>
    </Collapsible>
  );
};
