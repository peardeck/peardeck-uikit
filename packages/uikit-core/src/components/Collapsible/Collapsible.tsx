import React, { useState } from "react";
import styled from "styled-components";
import { CollapsibleHeader, CollapsibleHeaderProps } from "./CollapsibleHeader";

export interface CollapsibleProps extends CollapsibleHeaderProps {
  /**
   * Called when the collapsed state should be changed.
   *
   * Signature:
   * ```
   * onCollapsedChanged(shouldBeCollapsed: boolean)
   * ```
   */
  onCollapsedChange: Function;

  /**
   * The content to render in the collapsible content section.
   */
  children: React.ReactNode;
}

const StyledCollapsible = styled.div`
  border: 1px solid #eee;
`;

interface StyledCollapsibleContentProps {
  collapsed: boolean;
}

const StyledCollapsibleContent = styled.div<StyledCollapsibleContentProps>`
  transition: all 0.5s;
  overflow: hidden;
  max-height: ${({ collapsed }: { collapsed: boolean }): string =>
    collapsed ? "0" : "600px"};
`;

/**
 *  Provides a container with a clickable header that will collapse and expand child content.
 */
export const Collapsible = ({
  children,
  collapsed,
  onCollapsedChange,
  ...props
}: CollapsibleProps): JSX.Element => (
  <StyledCollapsible {...props}>
    <CollapsibleHeader
      title="The Title"
      subTitle="A subtitle goes here"
      collapsed={collapsed}
      onCollapsedChange={onCollapsedChange}
    />
    <StyledCollapsibleContent collapsed={collapsed}>
      {children}
    </StyledCollapsibleContent>
  </StyledCollapsible>
);

interface UncontrolledCollapsibleProps {
  /**
   * Test!
   */
  title: string;
  /* Subtitle info goes here? */
  subTitle?: string;
  children: React.ReactNode;
}

export const UncontrolledCollapsible = (
  props: UncontrolledCollapsibleProps
): JSX.Element => {
  const [collapsed, onCollapsedChange] = useState(false);
  return (
    <Collapsible
      {...props}
      collapsed={collapsed}
      onCollapsedChange={onCollapsedChange}
    />
  );
};
