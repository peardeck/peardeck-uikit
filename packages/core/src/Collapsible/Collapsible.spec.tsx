import { create } from "react-test-renderer";
import React from "react";
import { Collapsible } from "./Collapsible";

describe("Collapsible", () => {
  test("is a named export", () => {
    expect(Collapsible).toBeDefined();
  });

  describe("collapsed", () => {
    test("it matches the snapshot", () => {
      const component = create(
        <Collapsible
          collapsed
          onCollapsedChange={() => {}}
          subTitle="To the collapsed test"
          title="Welcome"
        >
          Here is some content for you
        </Collapsible>
      );
      expect(component.toJSON()).toMatchSnapshot("collapsed");
    });
  });

  describe("expanded", () => {
    test("it matches the snapshot", () => {
      const component = create(
        <Collapsible
          collapsed={false}
          onCollapsedChange={() => {}}
          subTitle="To the expanded test"
          title="Welcome"
        >
          Here is some content for you
        </Collapsible>
      );
      expect(component.toJSON()).toMatchSnapshot("expanded");
    });
  });
});
