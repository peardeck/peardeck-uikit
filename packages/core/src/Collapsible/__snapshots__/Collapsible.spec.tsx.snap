// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapsible collapsed it matches the snapshot: collapsed 1`] = `
.c1 {
  color: #547272;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 0;
  background-color: transparent;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: 100%;
  text-align: left;
  margin: 0;
  padding: 0.5em;
  cursor: pointer;
}

.c1:focus {
  outline: 0;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c3 {
  font-size: 2em;
  text-transform: uppercase;
}

.c4 {
  font-size: 1.25em;
}

.c5 {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
  opacity: 0.7;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c0 {
  border: 1px solid #eee;
}

.c6 {
  -webkit-transition: all .5s;
  transition: all .5s;
  overflow: hidden;
  max-height: 0;
}

<div
  className="c0"
  title="Welcome"
>
  <button
    className="c1"
    onClick={[Function]}
  >
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        The Title
      </div>
      <div
        className="c4"
      >
        A subtitle goes here
      </div>
    </div>
    <div
      className="c5"
    >
      <svg
        aria-hidden="true"
        focusable="false"
        height="1em"
        tabIndex={-1}
        viewBox="0 0 63.34 36.59"
        width="1em"
      >
        <title />
        <path
          d="M31.77 0a8.86 8.86 0 0 0-6.38 2.62l-9.42 9.3L1.81 26.08a5.87 5.87 0 0 0-.14 8.3l.18.18a6.85 6.85 0 0 0 9.3 0l7.28-6.87 7.24-6.67 6.13-6.07 5.87 6.07 7.25 6.88 7.31 6.85a6.85 6.85 0 0 0 9.3 0 5.89 5.89 0 0 0 .14-8.31l-.15-.15-14-14.12-9.3-9.3A9.06 9.06 0 0 0 31.77 0z"
        />
      </svg>
    </div>
  </button>
  <div
    className="c6"
  >
    Here is some content for you
  </div>
</div>
`;

exports[`Collapsible expanded it matches the snapshot: expanded 1`] = `
.c1 {
  color: #547272;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 0;
  background-color: transparent;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: 100%;
  text-align: left;
  margin: 0;
  padding: 0.5em;
  cursor: pointer;
}

.c1:focus {
  outline: 0;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c3 {
  font-size: 2em;
  text-transform: uppercase;
}

.c4 {
  font-size: 1.25em;
}

.c5 {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  opacity: 0.7;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c0 {
  border: 1px solid #eee;
}

.c6 {
  -webkit-transition: all .5s;
  transition: all .5s;
  overflow: hidden;
  max-height: 600px;
}

<div
  className="c0"
  title="Welcome"
>
  <button
    className="c1"
    onClick={[Function]}
  >
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        The Title
      </div>
      <div
        className="c4"
      >
        A subtitle goes here
      </div>
    </div>
    <div
      className="c5"
    >
      <svg
        aria-hidden="true"
        focusable="false"
        height="1em"
        tabIndex={-1}
        viewBox="0 0 63.34 36.59"
        width="1em"
      >
        <title />
        <path
          d="M31.77 0a8.86 8.86 0 0 0-6.38 2.62l-9.42 9.3L1.81 26.08a5.87 5.87 0 0 0-.14 8.3l.18.18a6.85 6.85 0 0 0 9.3 0l7.28-6.87 7.24-6.67 6.13-6.07 5.87 6.07 7.25 6.88 7.31 6.85a6.85 6.85 0 0 0 9.3 0 5.89 5.89 0 0 0 .14-8.31l-.15-.15-14-14.12-9.3-9.3A9.06 9.06 0 0 0 31.77 0z"
        />
      </svg>
    </div>
  </button>
  <div
    className="c6"
  >
    Here is some content for you
  </div>
</div>
`;
